### ISSUE TITLE
Self play random count problem

### ISSUE THREAD

**Issue by kennychenfs** [gcp/leela-zero#2102]
Hello, gcp.
I found autogtp v16 uses -m 30.
But v17 uses -m 999.
Is this just a mistake or a special purpose?

**Comment by Ishinoshita** [comment]
@kennychenfs IIRC, self-play games parameters are not defined by autogtp itself but sent by leele-zero-server, when autogtp connect to the server to request a new game.

Seems like no change have been implemented yet respect to --randomcount parameter (-m 30 vs -m 999), per server.js:
```
const options = { playouts: "0", visits: "3201", resignation_percent: "5", noise: "true", randomcnt: "999" };
if (Math.random() < 0.2) options.resignation_percent = "0";
```
*Not sure I read the code well, however.*

**Comment by kennychenfs** [comment]
But I found difference form
http://zero.sjeng.org/self-plays
Those are really different.
V17
http://zero.sjeng.org/view/2a11cea301f4e370b00d44ce8dc63e5de8886e4f7ad3528c4cdf799e96fba44b?viewer=wgo
V16
http://zero.sjeng.org/view/5027adbac486e1af7db71856fd7cdc20e01e33970d618838325794bd75e12674?viewer=wgo
The lz option in Bottom right corner is different.

Ishinoshita <notifications@github.com> 於 2018年12月29日 週六 上午2:55寫道：

**Comment by alreadydone** [comment]
Yes probably you are seeing BRII producing -m 30 games with leelaz compiled earlier without version number bumped up.

**Comment by Friday9i** [comment]
My computer produced this selfplay game, adjudicated after 2 successive pass at move 69 and 70?!? Strange, isn't it? Here it is: 
[95dfc04abe1b420fb4bd44901898c800.zip](https://github.com/gcp/leela-zero/files/2729953/95dfc04abe1b420fb4bd44901898c800.zip)

**Comment by Vandertic** [comment]
Probably black passed because of -m 999 and white then passed because it would win.

**Comment by Ishinoshita** [comment]
@Friday9i @Vandertic  Indeed, I have observed that in many LZ training data chuncks, and although there is supposed to be a minimum of 92 moves before resign is allowed, there are always some games finished *before* 92 moves, as early as to 60-70 moves counts.

The fraction of games resigned at 92 moves is much over-represenred in the game length distribution, due to the 92 move count rule. This indicates a substantial fraction of games have already hit the 5% resign threshold when reaching 92 moves count. I haved observed this happening as early as 40 moves count (e.g. big blunder at move 20, win rate already below 5% at move 40). 

Below 5% wr, search policy becomes much flat, noise effect is amplified comparatively and can push Pass move search policy high enough to be picked. So, I guess this explains the observed fraction of games finished before 92 moves.
